editions/tw5.com/tiddlers/widgets/CodeblockWidget.tid

```diff
@@ -0,0 +1,34 @@
+caption: codeblock
+created: 20151103160200000
+modified: 20151103160200000
+tags: Widgets
+title: CodeBlockWidget
+type: text/vnd.tiddlywiki
+
+! Introduction
+
+The ''codeblock'' widget renders text in `<pre>` and `<code>` blocks, causing it to be
+displayed monospace. A language may optionally be specified using the
+''language'' attribute, however syntax highlighting will only be used if the
+[[Highlight Plugin]] is installed.
+
+! Content and Attributes
+
+The content of the `<$codeblock>` widget is ignored.
+
+|!Attribute |!Description|
+|code|Contents of the block to render as code|
+|language|Programming language for syntax highlighting|
+
+! Examples
+
+Here is an example embedding the contents of a tiddler as a code block.
+
+<$macrocall $name='wikitext-example-without-html' 
```

Comment:
```
<$button set=!!test setTo="""<$macrocall $name=".js-macro-link" text={{!!test}}/>""">
Test button</$button>
```

instead of

```
<$button set=!!test setTo=<$macrocall $name=".js-macro-link" text={{!!test}}/> >
Test button</$button>
```